File: backend/app/schemas/session.py
from pydantic import BaseModel

PLACEHOLDER_TITLE = "新会话"
TITLE_SOURCE_PLACEHOLDER = "placeholder"
TITLE_SOURCE_AUTO = "auto"
TITLE_SOURCE_USER = "user"


class Session(BaseModel):
    id: str
    created_at: int
    updated_at: int
    workspace_path: str | None = None
    title: str = PLACEHOLDER_TITLE
    title_source: str = TITLE_SOURCE_PLACEHOLDER


class SessionCreate(BaseModel):
    workspace_path: str


class SessionRebind(BaseModel):
    workspace_path: str


class SessionRename(BaseModel):
    title: str


class ToolCallFunction(BaseModel):
    name: str
    arguments: str


class ToolCall(BaseModel):
    id: str
    type: str = "function"
    function: ToolCallFunction


class ChatMessage(BaseModel):
    role: str
    content: str | None = None
    tool_calls: list[ToolCall] | None = None
    tool_call_id: str | None = None
    timestamp: int | None = None


class SessionHistory(BaseModel):
    session_id: str
    messages: list[ChatMessage]
    context_usage: dict | None = None
    api_usage: dict | None = None
    session_token_stats: dict | None = None
    todos: list[dict] | None = None

File: frontend/src/api/session.ts
import api from './index'
import type { ApiUsage, ContextUsage, SessionTokenStats } from './chat'

export interface Session {
  id: string
  created_at: number
  updated_at: number
  workspace_path?: string | null
  title: string
  title_source?: 'placeholder' | 'auto' | 'user'
}

// OpenAI 标准消息格式
export interface ToolCallFunction {
  name: string
  arguments: string  // JSON 字符串
}

export interface ToolCall {
  id: string
  type: 'function'
  function: ToolCallFunction
}

export interface ChatMessage {
  role: 'user' | 'assistant' | 'tool'
  content?: string
  tool_calls?: ToolCall[]
  tool_call_id?: string
  timestamp?: number
}

export interface SessionHistory {
  session_id: string
  messages: ChatMessage[]
  context_usage?: ContextUsage | null
  api_usage?: ApiUsage | null
  session_token_stats?: SessionTokenStats | null
  todos?: Array<{ id: string; content: string; status: string }> | null
}

export const sessionApi = {
  list: async () => {
    return api.get<{ sessions: Session[] }>('/session/list')
  },
  create: async (workspacePath: string) => {
    return api.post<{ session_id: string }>('/session/create', {
      workspace_path: workspacePath,
    })
  },
  rebind: async (id: string, workspacePath: string) => {
    return api.post<Session>(`/session/${id}/workspace`, {
      workspace_path: workspacePath,
    })
  },
  rename: async (id: string, title: string) => {
    return api.patch<Session>(`/session/${id}/title`, { title })
  },
  get: async (id: string) => {
    return api.get<Session>(`/session/${id}`)
  },
  delete: async (id: string) => {
    return api.delete(`/session/${id}`)
  },
  getHistory: async (id: string) => {
    return api.get<SessionHistory>(`/session/${id}/history`)
  },
}
